lib/libzfs/libzfs_import.c

```diff
@@ -1675,6 +1675,45 @@ zpool_clear_label(int fd)
 	return (0);
 }
 
+/*
+ * Scan a list of devices for zfs devices.
+ */
+static int
+zpool_find_import_path(libzfs_handle_t *hdl, kmutex_t *lock,
+    avl_tree_t **slice_cache, char **path, int paths)
+{
+	avl_tree_t *cache;
+	rdsk_node_t *slice;
+	int i, error;
+
+	*slice_cache = NULL;
+	cache = zfs_alloc(hdl, sizeof (avl_tree_t));
+	avl_create(cache, slice_cache_compare, sizeof (rdsk_node_t),
+	    offsetof(rdsk_node_t, rn_node));
+
+	for (i = 0; i < paths; i++) {
+		slice = zfs_alloc(hdl, sizeof (rdsk_node_t));
+		error = asprintf(&slice->rn_name, "%s", path[i]);
+		if (error == -1) {
+			free(slice);
+			continue;
+		}
+		slice->rn_vdev_guid = 0;
+		slice->rn_lock = lock;
+		slice->rn_avl = cache;
+		slice->rn_hdl = hdl;
+		slice->rn_order = i + IMPORT_ORDER_SCAN_OFFSET;
+		slice->rn_labelpaths = B_FALSE;
+		mutex_enter(lock);
```

Comment:
Should be changed to `pthread_mutex_lock` post c25b8f99f8dcbe898b81728e6a9dab107df4fc0b cleanup.